bin/fetch-proxy

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# If the first argument to this script is "latest" or unset, it fetches the
+# latest proxy binary via build.l5d.io/linkerd2-proxy/latest.txt. If it's set
+# to a 7-character commit sha from the master branch of the linkerd2-proxy repo,
+# it will fetch that binary instead.
```

Comment:
We don't actually test that it's a 7-character commit sha, and I don't think we should. I would just leave out this level of detail from the comment and refer to it as a version string or something